### PR TITLE
Draft: Convert D01 up until L470

### PR DESCRIPTION
**Pull request by jr-mandy**
This PR attempts to convert existing SPSS syntax available in [2021-22/D01 Make Individual File.sps](https://github.com/Public-Health-Scotland/source-linkage-files/blob/main-R/2021-22/D01%20Make%20Individual%20File.sps) up until L470.

I have put all new functions in a single script (create_individual_file.R) as I didn't want to spend any time guessing your preferred split of functions into files and naming conventions etc.

The top-level function is `create_individual_file()` which I expect to add more steps to as I continue converting the SPSS syntax.

The bulk of the converted syntax involved creating a large number of columns based on a variety of conditions. Using tidyverse-style column creation this was not a very pretty task, but I've tried to generalise where possible and avoid having to copy-paste any one condition over and over again, so all top-level conditions are passed into associated functions in the `add_all_columns()` function (definitely feel free to rename). 

Overall, I didn't want to spend too much time trying to understand what all the acronyms meant, so a lot of the function documentation is quite brief (especially in terms of background context which I lack) - so this might be something that you want to flesh out more. I also didn't want to spend too much time just looking through all existing functions etc., so there may very well be bits of code that could be replaced with existing PHS-style helper functions, so that might be another thing to look out for.

Finally, I have tried checking my output against the SPSS output, but this is probably worth you having a double check :)

### CODE REVIEW

**Comment by jr-mandy** [comment]
@Moohan 
Here is the first PR! I've left it for you to decide about the spell check and whether to rebase/merge with main etc.
Also it's up to you if you want to do PR ping pong and assign back to me with comments/suggestions or whether you want to just implement any changes yourself? I suppose any time I spend tweaking takes away available time for converting further SPSS so up to you how we prioritise :)

File: R/create_individual_file.R [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+
+#' Remove blank CHI
+#'
+#' @description Convert blank strings to NA and remove NAs from CHI column
+#'
+#' @inheritParams create_individual_file
+remove_blank_chi <- function(episode_file) {
+  episode_file %>%
+    dplyr::mutate(chi = dplyr::na_if(.data$chi, "")) %>%
+    dplyr::filter(!is.na(.data$chi))
```
Comment (shintoLamp) [inline R/create_individual_file.R:23]: dplyr::filter(!is_missing(.data$chi))

File: R/create_individual_file.R [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+    add_cmh_columns("CMH", .data$recid == "CMH") %>%
+    add_dd_columns("DD", .data$recid == "DD") %>%
+    add_nsu_columns("NSU", .data$recid == "NSU") %>%
+    add_nrs_columns("NRS", .data$recid == "NRS") %>%
+    add_hl1_columns("HL1", .data$recid == "HL1") %>%
+    add_ch_columns("CH", .data$recid == "CH") %>%
+    add_hc_columns("HC", .data$recid == "HC") %>%
+    add_at_columns("AT", .data$recid == "AT") %>%
+    add_sds_columns("SDS", .data$recid == "SDS")
+}
```
Comment (shintoLamp) [inline R/create_individual_file.R:108]: Need to standardise prefixes to be in lowercase for R output

**Review by shintoLamp — COMMENTED** [review]
Could do with some unit tests

**Comment by Moohan** [comment]
This branch has now been merged into the #561 and is no longer needed.

**Pull request closed without merging**